src/components/webview/MainProvider.ts

```diff
@@ -29,10 +30,40 @@ import {
 	CodemodNodeHashDigest,
 	selectCodemodArguments,
 } from '../../selectors/selectCodemodTree';
-import { isNeitherNullNorUndefined } from '../../utilities';
+import { buildTypeCodec, isNeitherNullNorUndefined } from '../../utilities';
 
 const X_INTUITA_ACCESS_TOKEN = 'X-Intuita-Access-Token'.toLocaleLowerCase();
 
+export const validateAccessToken = async (
+	accessToken: string,
+): Promise<boolean> => {
+	const result = await axios.post(
+		'https://telemetry.intuita.io/validateAccessToken',
+		{},
+		{
+			headers: {
+				[X_INTUITA_ACCESS_TOKEN]: accessToken,
+			},
```

Comment:
please add a timeout to this request (I think 5s is the maximum time that makes sense here)>

Reply:
addressed